pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -111,7 +111,14 @@ def on_train_start(self, trainer, pl_module) -> None:
                 f' since gpus attribute in Trainer is set to {trainer.gpus}.'
             )
 
-        self._gpu_ids = ','.join(map(str, trainer.data_parallel_device_ids))
+        self._device_ids = list(trainer.data_parallel_device_ids)
+
+        cuda_visible_devices = os.getenv('CUDA_VISIBLE_DEVICES', default=None)
+        if cuda_visible_devices is None:
+            self._gpu_ids = ','.join(map(str, self._device_ids))
```

Comment:
you could directly define this as the default for cuda_visible_devices, so the mapping code below simplifies and there should be no need for if-else .)

Reply:
Should I `import cuda` and use `default=range(torch.cuda.device_count())` here? It cannot be assgined as `default=','.join(map(str, self.device_ids))`.

```python
cuda_visible_devices = default = ','.join(map(str, device_ids))
cuda_visible_devices = cuda_visible_devices.split(',')
gpu_ids = ','.join(map(cuda_visible_devices.__getitem__, device_ids))
```

- device_ids = [1, 2, 0] -> gpu_ids = [2, 0, 1]
- device_ids = [1, 2] -> IndexError

Reply:
What I meant is, you can do something like this:

```python
 cuda_visible_devices = os.getenv('CUDA_VISIBLE_DEVICES', ','.join(map(str, self._device_ids)))
 cuda_visible_devices = cuda_visible_devices.split(',')
 gpu_ids = ','.join(map(cuda_visible_devices.__getitem__, self._device_ids))
```

Reply:
Yep, something like this?
```py

cuda_visible_devices = os.getenv('CUDA_VISIBLE_DEVICES', ','.join(range(torch.cuda.device_count()))
 cuda_visible_devices = cuda_visible_devices.split(',')
 gpu_ids = ','.join(map(cuda_visible_devices.__getitem__, self._device_ids))
```

Reply:
Also, it's probably best if we keep `gpu_ids` a list internally, like device_ids.